modules/jax/src/main/ts/ephox/jax/core/HttpError.ts

```diff
@@ -10,10 +12,10 @@ export const enum HttpErrorCode {
   InternalServerError = 500
 }
 
-export interface HttpError {
-  message: string;
-  status: HttpErrorCode;
-  responseText: string;
+export interface HttpError<T extends ResponseType> {
+  readonly message: string;
+  readonly status: HttpErrorCode;
+  readonly responseText: ResponseTypeMap[T];
 }
 
-export const httpError = <T>(status: number, message: string, responseText: string): Result<T, HttpError> => Result.error<T>({ message, status, responseText });
+export const httpError = <T extends ResponseType>(status: number, message: string, responseText: T): Result<T, HttpError<T>> => Result.error<T>({ message, status, responseText });
```

Comment:
```suggestion
export const httpError = <T extends ResponseType>(status: number, message: string, responseText: ResponseTypeMap[T]): Result<T, HttpError<T>> => Result.error<T>({ message, status, responseText });
```
Should it be this instead?